lib/propolis/src/util/sys.rs

```diff
@@ -7,8 +7,8 @@ use std::io::ErrorKind;
 // Deal with libc bits which vary enough between OSes to make cargo-check a pain
 
 #[cfg(target_os = "illumos")]
-pub fn ioctl<T>(fd: RawFd, cmd: i32, data: *mut T) -> Result<i32> {
-    let res = unsafe { libc::ioctl(fd, cmd, data) };
+pub unsafe fn ioctl<T>(fd: RawFd, cmd: i32, data: *mut T) -> Result<i32> {
```

Comment:
Update the not-illumos variant to match

Reply:
This is already done in the `psl` branch where I'm staging the squashed changes from this PR.